Updating WebDriver actions tests to be IE-friendly

jimevans opened the pull request:
Removing ES6 constructs from tests and supporting test page. Also removing
validation of keyboard event `code` property, as IE (and Edge) do not
support it.

<!-- Reviewable:start -->

<!-- Reviewable:end -->

jimevans committed:
Updating WebDriver actions tests to be IE-friendly

Removing ES6 constructs from tests and supporting test page. Also removing
validation of keyboard event `code` property, as IE (and Edge) do not
support it.

w3c-bots commented:
# Build PASSED

Started: 2017-10-19 20:45:33
Finished: 2017-10-19 21:27:07


<h2>Failing Jobs</h2>
<ul>

<li>chrome:unstable</li>

<li>safari:10.0</li>

<li>MicrosoftEdge:14.14393</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/19716)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/290155688)

andreastt commented:
I feel @mjzffr’s input is required here.

jgraham commented:
@mjzffr Want to review this?

jgraham commented:
FWIW I think this change is probably fine. I'd also be happy, maybe happier, if we just didn't require any properties that were undefined in the event to be present so that we could keep the stricter checks for browsers implementing `code`.

mjzffr commented on webdriver/tests/actions/support/test_actions_wdspec.html:
```diff
@@ -132,27 +145,31 @@
           els.dragArea.addEventListener("mousemove", moveHandler);
-          els.dragArea.addEventListener("mouseup", drop(moveHandler), {once: true});
+          initialMoveHandler = drop(moveHandler);
+          els.dragArea.addEventListener("mouseup", dropOnce);
```
I'd like to avoid the global `initialMoveHandler`. Maybe define `dropOnce` inside `grab`, then you shouldn't need `initialMoveHandler`.

jimevans replied:
I've gone as far as I can trying to eliminate the global event handler. If someone would show me the way to do that, I'd be happy to oblige.

jimevans committed:
Restore 'code' verification for key actions for browsers supporting it

jimevans committed:
Making removal of 'code' element more efficient

Also updated the keyboard actions tests to clear the event log in the test
page after clicking on the key input element, but before sending the
keystrokes.

jimevans committed:
Removing global event handler reference

andreastt reviewed (approved):
This looks good to me. It also passes the tests.
